src/psyclone/psyir/transformations/acc_update_trans.py

```diff
@@ -269,40 +260,22 @@ def _add_update_directives(self, node_list):
                 # leverage overlapping communication and computation.
                 update_pos = sched.children.index(child) + node_offset
 
-                if idx == IN:
+                if acss_type == IN:
                     beg, end = None, update_pos
-                elif idx == OUT:
+                elif acss_type == OUT:
                     beg, end = update_pos, None
 
-                # The (in)ability to place variables in update directives in
-                # outer schedules is determined by the (in)existence of textual
-                # and loop-carried dependencies in the current schedule.
-                text_dep_stmts, text_sync = sched.children[beg:end], set()
-                loop_dep_stmts, loop_sync = sched.children[end:beg], set()
-
-                for dep_stmts, sync in [(text_dep_stmts, text_sync),
-                                        (loop_dep_stmts, loop_sync)]:
-                    kern_sig = set()
-                    for stmt in dep_stmts:
-                        for acc in stmt.walk(self._acc_compute):
-                            # Kernel outputs are potential both input and
-                            # output dependencies. The latter is because we
-                            # must guarantee no kernel write is overwritten by
-                            # an earlier host write whose respective update
-                            # device directive could appear later.
-                            kern_sig.update(acc.out_kernel_references)
-                            if idx == OUT:
-                                kern_sig.update(acc.in_kernel_references)
-                    # Amongst the variables touched in the host region, find
-                    # those that need synchronisation at this schedule.
-                    sync.update(host_sig.intersection(kern_sig))
-                    # If there is a statement (e.g. a call) in the current
-                    # schedule that may launch device kernels touching data
-                    # which is also touched in this host region, we
-                    # conservatively insert update directives in this schedule
-                    # for all the variables involved regardless of direction.
-                    if any(stmt.walk(self._brk_nodes) for stmt in dep_stmts):
-                        sync.update(host_sig)
+                # The inability to place variables in update directives in
+                # outer schedules is determined by their involvement in textual
```

Comment:
I think this is looking better now - thank you :-) Please could you clarify/re-word what is meant by "textual dependence" (both here and in earlier and later comments). I've googled for it and am non the wiser.

Reply:
Oh! That's crucial to understanding what's happening, thank you for letting me know, does it make more sense now?

Reply:
Yes, thanks, it does. It's a relief that it means what I was assuming it meant. "within no loop" sounds a bit strange though, do you mean "not within a loop"?

Reply:
Done.